### PR TITLE
add regnet, res2net, resnest, sknet encoders

### PR DESCRIPTION
**Pull request by azkalot1**
Integrating some encoders using https://github.com/rwightman/pytorch-image-models

### CODE REVIEW

**Comment by azkalot1** [comment]
Build fails since batchnorm in some of the encoders requires >= 2 samples in the batch. 
I suggest test can be changes from 
```
def get_sample(model_class):
    if model_class in [smp.Unet, smp.Linknet, smp.FPN, smp.PSPNet, smp.UnetPlusPlus]:
        sample = torch.ones([1, 3, 64, 64])
    elif model_class == smp.PAN:
        sample = torch.ones([2, 3, 256, 256])
    elif model_class == smp.DeepLabV3:
        sample = torch.ones([2, 3, 128, 128])
    else:
        raise ValueError("Not supported model class {}".format(model_class))
    return sample
```
to simply

```
def get_sample(model_class):
    if model_class in [smp.Unet, smp.Linknet, smp.FPN, smp.PSPNet, smp.UnetPlusPlus,  smp.PAN,  smp.DeepLabV3]:
        sample = torch.ones([2, 3, 256, 256])
    else:
        raise ValueError("Not supported model class {}".format(model_class))
    return sample
```

**Comment by qubvel** [comment]
It is better to keep sample as small as possible, because it will increase test time significantly (only cpu is available). May be it is better to add one more 'if' case.

**Comment by qubvel** [comment]
Or set batchnorms to eval mode :)

**Comment by azkalot1** [comment]
Should I just use `model.eval()`?

**Comment by JulienMaille** [comment]
@azkalot1 May I ask you for feedback on the potential caveats of #373 ? Thanks
